content/blog/end-to-end-easy-to-use-pipeline-for-training-a-model-on-medmnist-v2-using-hpe-machine-learning-development-environment-flask.md

```diff
@@ -0,0 +1,514 @@
+
+#### Model Sharing and Reproducibility 
+
+The HPE Machine Learning Development Environment WebUI makes it easy to track experiments and see which model configurations resulted in particular results, across a team. This makes reproducibility easy, something that is of utmost importance when developing models for a use case like predicting cancer in biomedical images.  
+
+### Appendix: Building an E2E solution 
+
+#### Step 1: Model Training 
+
+To get started, we’ll take a look at the original training script provided [here](https://github.com/MedMNIST/experiments/blob/main/MedMNIST2D/train_and_eval_pytorch.py). 
+
+This script contains all the functionality needed to download the dataset as well as train and test the model (a commonly used convolutional neural network architecture – ResNet), on the PathMNIST data. 
+
+The original script has some boilerplate code that can be removed since HPE Machine Learning Development Environment handles functionality such as distributed training and experiment visualization. For a full guide to model porting, refer to the [model porting guide](https://hpe-mlde.determined.ai/latest/tutorials/pytorch-porting-tutorial.html#pytorch-porting-tutorial). Here, we’ll walk through the steps taken to port the code to HPE Machine Learning Development Environment using the PyTorch API. The PyTorch API is a high-level API which allows you to utilize the full functionality of HPE Machine Learning Development Environment out-of-the-box. 
+
+#### Step 1.1: Connect to HPE Machine Learning Development Environment  
+
+Once your admin has provisioned a cluster with HPE Machine Learning Development Environment and configured you as a user, connect to HPE Machine Learning Development Environment by exporting the DET_MASTER variable: 
+
+```yaml
+export DET_MASTER=<your cluster address>
+```
+
+Then log in as your user: 
+
+```yaml
+det login user <your username>
+```
+
+You can check to make sure you are logged in via 
+
+```yaml
+det user whoami
+```
+
+Refer to the Determined User Guide and the Reference Page for more information on how to interact with the cluster.  
+
+*In steps 1.2 – 1.4, we’re going to describe how to port this model to HPE Machine Learning Development Environment step-by-step, (model_def.py). If you are interested in running the final training job, skip to step 1.5.* 
+
+#### Step 1.2: Port Model Definition  
+
+To train your own custom model using one of Determined’s high level APIs, such as the PyTorch API, you need to port your code to the API first. We provide a template of all the functions Determined needs to run your training loop [here](https://docs.determined.ai/latest/training/apis-howto/api-pytorch-ug.html#pytorch-trial). Fill them out one-by-one to port your code. Once these functions are populated, Determined can use these along with the provided configuration file, to run your experiment. 
```

Comment:
Instead of "We provide a template of..." change it to "A template of all the functions Determined needs to run your training loops is provided here."

In the following sentence, the phrase "one-by-one" should be enclosed with commas so it would read:

"Fill them out, one-by-one, to port your code."

In the last sentence, you need one more comma just prior to "along with". So the entire sentence would read:

"Once these functions are populated, Determined can use these, along with the provided configuration file, to run your experiment."